sw/airborne/modules/computer_vision/lib/vision/image.c

```diff
@@ -736,8 +736,8 @@ void image_show_flow_color(struct image_t *img, struct flow_t *vectors, uint16_t
       .y = vectors[i].pos.y / subpixel_factor
     };
     struct point_t to = {
-      .x = (vectors[i].pos.x + vectors[i].flow_x) / subpixel_factor,
-      .y = (vectors[i].pos.y + vectors[i].flow_y) / subpixel_factor
+      .x = (uint32_t)roundf((float)(vectors[i].pos.x + vectors[i].flow_x) / subpixel_factor),
+      .y = (uint32_t)roundf((float)(vectors[i].pos.y + vectors[i].flow_y) / subpixel_factor)
```

Comment:
Not sure what you mean here. Adding the cast where I put it will case the result of the sum of pos and flow to float before the divide by subpixel_factor and subsequent round, exactly what I intended to do (if I am not mistaken). 

The result is indeed implicitly cast but I like to add the explicit cast to highlight that a cast is happing in the assignment.

Reply:
You are adding an int32_t (`vectors[i].flow_y`) to a uint32_t (`vectors[i].pos.y`) here. To perform the addition, `vectors[i].flow_y` [is first cast to a uint32_t](https://stackoverflow.com/questions/19446888/adding-signed-and-unsigned-int), which is probably not what you want. You could fix it by casting `vectors[i].pos.y` to int32_t first, but since you want a float anyway you may as well cast it to that.

Reply:
If that were true, the result of the addition would overflow and the result would still be correct once flow is not larger than pos (which is why the code probably still works). 

But clearly this is not good. As this is done many times in the code it may be more prudent to convert the position to signed instead of unsigned.

Reply:
Sounds good to me, int32 is large enough for the images we work with ;).